content/kb/using-streamlit/buttons.md

```diff
@@ -0,0 +1,412 @@
+---
+title: Button behavior and examples
+slug: /knowledge-base/using-streamlit/buttons
+---
+
+# Button behavior and examples
+
+## TLDR
+
+Buttons do not retain state. They return `True` on the page load resulting from
+their click and immediately return to `False` on the next load. If a displayed
+element is nested inside `if st.button('Click me'):`, the element will be
+visible when the button is clicked and disappear as soon as the user takes their
+next action. This is because the page reloads and the button becomes `False`.
+
+## Use cases for buttons
+
+When code is conditioned on a button's value, it will execute once in
+response to the button being clicked and not again (until the button is clicked
+again).
```

Comment:
Perhaps reword slightly or add some more text here? It's unclear why this sentence follows a heading that says use cases.